src/rcc.rs

```diff
@@ -418,6 +418,8 @@ impl RccExt for RCC {
 
         let ahb_freq = match cfgr.ahb_pre {
             AHBPrescaler::NotDivided => sys_clk,
+            // skip /32
+            pre if pre as u8 & 0b0100 > 0 => sys_clk / (1 << (pre as u8 - 6)),
```

Comment:
Won't this panic because `0b0100` - 6 will always underflow? I could be misunderstanding the logic here but it would be good to confirm this on real hardware if you haven't already.

Reply:
if it was just `0b01xx`, then yes.
But since except for the `NotDivided` enum kind, all of the `u8` values assigned to the enum kinds have the fourth bit (index 3) set, subtracting 6 is fine.
If you look closely, I'm actually subtracting _less_ here than in the third match branch, which was the original implementation.

Honestly, I wasn't sure how to implement this. My preferred solution would be to simply define a function on the enum type that returns the prescaler factor as an integer using a large match statement. That would be much easier to read, understand, and check for errors such as this.
But I didn't want to mix styles too much, so I went with this implementation instead.

Reply:
Thanks for the explanation and the test on hardware - much appreciated! Makes sense now :)